### DISCUSSION TITLE
Importing module running into an error after upgrade

### DISCUSSION THREAD

**Discussion by srikumarkp** [prisma/prisma#23673] [Q&A] [answered]
### Question

Have two prisma.schema files where it is connecting to two different databases (mysql)

After upgrading from 5.7 to the latest version or any version after 5.8 running to an error:
```
1> Import newly created clients for schema1 as shown below
import { Prisma } from '@prisma/client/performance';
**import { PrismaClient } from '@prisma/client/performance';**

Highlited line is throwing the error below:

{"date":"2024-03-31T00:28:04.761Z","level":"error","message":"uncaughtException","payload":"{\"code\":\"MODULE_NOT_FOUND\",\"path\":\"/home/sri/src/dashboard_data_api/node_modules/@prisma/client/package.json\"}"}
/home/sri/src/dashboard_data_api/dist/apps/api/libs/common/src/logger/logger.service.js:17
    throw error;
    ^

Error: Cannot find module '/home/sri/src/dashboard_data_api/node_modules/@prisma/client/performance'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1070:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1063:15)
    at resolveExports (node:internal/modules/cjs/loader:547:14)
    at Function.Module._findPath (node:internal/modules/cjs/loader:621:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1034:27)
    at Function.Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/home/sri/src/dashboard_data_api/libs/common/src/prisma/prisma-performance.service.ts:4:1)
    at Module._compile (node:internal/modules/cjs/loader:1233:14) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/sri/src/dashboard_data_api/node_modules/@prisma/client/package.json'
}
```

### Expected behavior (optional)

Module should be loaded with no errors.

### Information about Prisma Schema, Client Queries and Environment (optional)

**schema1**:
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/@prisma/client/performance
  previewFeatures = ["relationJoins"]
}

// generator dbml {
//   provider   = "prisma-dbml-generator"
//   outputName = "schema-performance.dbml"
// }

datasource db {
  provider = "mysql"
  url      = env("PERFORMANCE_DATABASE_URL")
}
```
......
**schema2**:
```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["relationJoins"]
}

// generator dbml {
//   provider   = "prisma-dbml-generator"
//   outputName = "schema-performance.dbml"
// }

datasource db {
  provider = "mysql"
  url      = env("META_DATABASE_URL")
}
```
```
- OS: Ubuntu 22
- Database: Mysql 8.0.3
- Node.js version: 20.5.1
- Prisma version - 5.9 or above 
```
Please provide any help or guidance - thanks in advance for the help.

**Comment by srikumarkpz** [comment]
[package.json](https://github.com/prisma/prisma/files/14813697/package.json) - generated package.json from latest 
[package.json](https://github.com/prisma/prisma/files/14813700/package.json)
prisma version from 5.7

Notice the files are drastically different across version.

**Comment by ludralph** [comment, marked as answer]
Hi @srikumarkp 👋 

Thank you for raising this question.

The problem is related to the custom `output` path for the Prisma client in the `schema.prisma` file.

In your case, you're generating the Prisma client in a subfolder of `@prisma/client`, which seems to be causing the issue.

A [workaround](https://github.com/prisma/prisma/issues/22979#issuecomment-1957331557) in a similar issue  is to generate the Prisma client in a different directory that doesn't include `@prisma/client`. For example, you could change your schema.prisma file to something like this:
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/@my-firm/client/test"
}
```
And then import the client like this:
```js
import { Prisma, PrismaClient } from '@my-firm/client/test';
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by srikumarkp** [reply]
Hi @ludralph Thanks for the prompt response. It is working after i tried your suggestion.

Really appreciate the quick help !! Thanks again.
